OMCompiler/SimulationRuntime/c/simulation/solver/newton_diagnostics.c

```diff
@@ -0,0 +1,1280 @@
+         else
+            wDiag[i][j] = 0;
+   }
+
+   // Get inverse matrix inv_wDiag of wDiag
+   double** inv_wDiag = getInvJacobian( q, wDiag);
+
+   // Calculate tmp3 = | inv_wDiag | * tmp2
+   // (q x q matrix) *  (q x q matrix) --> q x q matrix
+   for( i = 0; i < q; i++)
+      for( j = 0; j < q; j++)
+         inv_wDiag[i][j] = fabs(inv_wDiag[i][j]);
+
+   double** tmp3 = MatMult( q, q, q, inv_wDiag, tmp2);
+
+   // Calculate Sigma = tmp3 * wDiag =  | inv_wDiag | * tmp2 * wDiag = | inv_wDiag | * -inv_fx * H_i * wDiag
+   double** Sigma = MatMult( q, q, q, tmp3, wDiag);
+
+   // Free dynamically allocated memory
+   for( i = 0; i < m; i++)
+   {
+      free(inv_fx[i]);
+      free(H_i[i]);
+      free(tmp1[i]);
+   }
+   free(inv_fx);
+   free(H_i);
+   free(tmp1);
+
+   for( i = 0; i < q; i++)
+   {
+      free(wDiag[i]);
+      free(inv_wDiag[i]);
+      free(tmp2[i]);
+      free(tmp3[i]);
+   }
+   free(wDiag);
+   free(inv_wDiag);
+   free(tmp2);
+   free(tmp3);
+
+   return Sigma;
+}
+
+// --------------------------------------------------------------------------------------------------------------------------------
+
+void PrintResults( DATA* data, unsigned sysNumber, unsigned m, unsigned p, unsigned q, unsigned* n_idx, unsigned* w_idx,
+                   double* x0, double* alpha, double*** Gamma_ijk, double** Sigma_ij)
+{
+   printf("  \n   ========================================================");
+   printf("  \n   Final results ");
+   printf("  \n   ========================================================\n\n");
+
+   NONLINEAR_SYSTEM_DATA* systemData = &(data->simulationInfo->nonlinearSystemData[sysNumber]);
+
+   unsigned i, j, k;
+   double eps = 1e-2;
+
+   // ----------------------------------------------- Debug -------------------------------------------------
+   /*printf("      Equations\n");
+   for( i = 0; i < p; i++)
+      printf("\n      i =%2d: %s", n_idx[i]+1, "???"); //, modelInfoGetFunction(&data->modelData->modelDataXml, i).name);
+   printf("\n\n");
+
+   printf("      Variables, initial guesses\n");
+   for( j = 0; j < q; j++)
+      printf("\n      j =%2d: %8s = %10.8g", w_idx[j]+1, data->modelData->realVarsData[var_id(w_idx[j], data, systemData)].info.name, x0[w_idx[j]]);
+   printf("\n\n");*/
+   // ------------------------------------------- end of Debug ----------------------------------------------
+
+   // Print alpha, Gamma, and Sigma if value > eps
+   // --------------------------------------------
+
+   printf("      alpha_i > %4.0e\n", eps);
+   printf("      ------------------------");
+   for( i = 0; i < p; ++i)
+   {
+      if (alpha[i] > eps)
+      {
+         if (alpha[i] < 1.e5)
+            printf("\n      alpha_%d =  %8.3f", n_idx[i]+1, alpha[i]);
+         else
+            printf("\n      alpha_%d =  %8.2e", n_idx[i]+1, alpha[i]);
+      }
+   }
+   printf("\n\n");
+
+   printf("      Gamma_ijk > %4.0e\n", eps);
+   printf("      ------------------------");
```

Comment:
Same here, use `infoStreamPrint`.

You can indent the next message with
```C
infoStreamPrint(LOG_NLS_NEWTON_DIAG, 1 /* indention level for next prints*/, "Stuff");
infoStreamPrint(LOG_NLS_NEWTON_DIAG, 0, "Indented stuff");
closeMessage(LOG_NLS_NEWTON_DIAG); // <-- important to close indention, otherwise XML logs mighty crash
```